opentimelineio/console/autogen_serialized_datamodel.py

```diff
@@ -0,0 +1,233 @@
+#!/uxr/bin/env python
+
+"""Generates documentation of the serialized data model for OpenTimelineIO."""
+
+import argparse
+import inspect
+import json
+import tempfile
+import sys
+
+try:
+    # python2
+    import StringIO as io
+except ImportError:
+    # python3
+    import io
+
+import opentimelineio as otio
+
+
+DOCUMENT_HEADER = """# OpenTimelineIO Serialized Data Documentation
+
+This document is a list of all the OpenTimelineIO classes that serialize to and
+from JSON, omitting SchemaDef plugins.
+
+This document is automatically generated by running
+ docs/autogen_serialized_datamodel.py, or by running `make doc-model`.  It is
+ part of the unit tests suite and should be updated whenever the schema changes.
+ If it needs to be updated, run: `make doc-model-update` and this file should be
+ regenerated.
+
+# Classes
+
+"""
+
+CLASS_HEADER = """
+### {}
+
+*full module path*: {}
+
+*documentation*:
+
+```
+{}
+```
+
+parameters:
+"""
```

Comment:
nit: I'd prefer to use the field names and keyword args to the `format` method when the templates are located away from where `format` is being called.